### PR TITLE
Add Darkmoon to Standard Sets

### PR DESCRIPTION
**Pull request by jk2112**
This should fix the secret tracker to display Darkmoon secrets.

### CODE REVIEW

**Comment by martinbonnin** [comment]
Hi ! Thanks for looking into this! While this will show the secrets, it will not detect them, you'll need to add the matching logic to [SecretLogic.kt](https://github.com/HearthSim/Arcane-Tracker/blob/957df42a0472cb2c501a73e1766e2f03df2d213e/kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt)

**Comment by jk2112** [comment]
I knew that was too easy! Here's a draft of the logic. The opponentWasDamaged variable is pretty rough, I'm open to feedback if there's a better way to do that using the entity, key, and tag objects.

**Comment by jk2112** [comment]
I'm a bit lost on this build error. Updating Gradle and editing build.gradle fixed the build on my local machine. Do you have to update Gradle on your build machine or is this something on my end?

File: kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt [hunk 1]
```diff
@@ -69,6 +70,9 @@ internal class SecretLogic(val cardJson: CardJson, val console: Console) {
                     exclude(game, CardId.COUNTERSPELL)
                     exclude(game, CardId.DIRTY_TRICKS)
 
+                    exclude(game, CardId.OH_MY_YOGG)
+                    exclude(game, CardId.OH_MY_YOGG1)
```
Comment (martinbonnin) [inline kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt:74]: That's an enchantmeny token card that shouldn't be needed

File: kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt [hunk 2]
```diff
@@ -185,6 +189,7 @@ internal class SecretLogic(val cardJson: CardJson, val console: Console) {
                     if (opponentHasRoomOnBoard(game)) {
                         exclude(game, CardId.BEAR_TRAP)
                         exclude(game, CardId.WANDERING_MONSTER)
+                        exclude(game, CardId.SHADOW_CLONE)
```
Comment (martinbonnin) [inline kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt:192]: Shadow Clone should only be excluded if the hero is attacked by a minion (unlike Wandering Monster where a hero attacking the hero would trigger). You can do this by wrapping the exclude in something like: `if(entity.tags[Entity.KEY_CARDTYPE] == Type.MINION`

**Comment by martinbonnin** [comment]
> Updating Gradle and editing build.gradle fixed the build on my local machine. Do you have to update Gradle on your build machine or is this something on my end?

That error happened because the `GOOGLE_PLAY_JSON` is a secret on github that's not available to pull request. This is to avoid someone opening a pull request and reading the secrets. I changed the CI scripts to use the mock in that case, hopefully that'll work better now 🤞 

> The opponentWasDamaged variable is pretty rough, I'm open to feedback if there's a better way to do that using the entity, key, and tag objects.

opponentWasDamaged looks legit to me. This has to be stateful in all cases 👍

**Comment by jk2112** [comment]
The Freezing Trap fix is for Issue #144, if that's too unrelated I can take it out and put it in its own PR.

File: kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt [hunk 3]
```diff
@@ -157,8 +155,10 @@ internal class SecretLogic(val cardJson: CardJson, val console: Console) {
         }
 
         if (attackingEntity.tags[Entity.KEY_CONTROLLER] == game.player?.entity?.PlayerID) {
-            // apparently, freezing trap will trigger even if the player hand is full
-            exclude(game, CardId.FREEZING_TRAP)
+            if(attackingEntity.tags[Entity.KEY_CARDTYPE] == Type.MINION) {
+                exclude(game, CardId.FREEZING_TRAP)
```
Comment (martinbonnin) [inline kotlin-hslog/src/commonMain/kotlin/net/hearthsim/hslog/parser/power/SecretLogic.kt:159]: 👍

**Comment by martinbonnin** [comment]
> The Freezing Trap fix is for Issue #144, if that's too unrelated I can take it out and put it in its own PR.

No problem at all, thanks for fixing this!

```
Undefined symbols for architecture x86_64:
  "_objc_setHook_setAssociatedObject", referenced from:
      _platform_objc_objc_setHook_setAssociatedObject_wrapper100 in libobjc-cache.a(result.o)
     (maybe you meant: knifunptr_platform_objc100_objc_setHook_setAssociatedObject
```

That's a new one... Never seen this before. Looks related to https://youtrack.jetbrains.com/issue/KT-43128

**Review by martinbonnin — APPROVED** [review]

**Comment by martinbonnin** [comment]
There's one last error on CI about signing release build which is another issue with github secrets. I'll fix this in master. Thanks for the contribution !

**Pull request merged**